ams/server/src/main/java/com/netease/arctic/server/dashboard/ServerTableDescriptor.java

```diff
@@ -206,4 +236,76 @@ public List<BaseMajorCompactRecord> getOptimizeInfo(String catalog, String db, S
       return record;
     }).collect(Collectors.toList());
   }
+
+  public List<PartitionBaseInfo> getTablePartition(ArcticTable arcticTable) {
+    if (arcticTable.spec().isUnpartitioned()) {
+      return new ArrayList<>();
+    }
+    Map<String, PartitionBaseInfo> partitionBaseInfoHashMap = new HashMap<>();
+    getTableFile(arcticTable, null, Integer.MAX_VALUE).forEach(e -> {
+      if (!partitionBaseInfoHashMap.containsKey(e.getPartitionName())) {
+        partitionBaseInfoHashMap.put(e.getPartitionName(), new PartitionBaseInfo());
+        partitionBaseInfoHashMap.get(e.getPartitionName()).setPartition(e.getPartitionName());
+      }
+      PartitionBaseInfo p = partitionBaseInfoHashMap.get(e.getPartitionName());
```

Comment:
p makes none sense, plz use a better name

Reply:
resolved in new commit